Branch1

ahmed-nad opened the pull request.

hamzaabbasi16 committed:
Add files via upload

hamzaabbasi16 committed:
Add files via upload

ahmed-nad commented on main.cpp:
```diff
@@ -10,28 +10,50 @@
+public:
+	void load_data_from_file(std::string file_name);
+	
+	float get_r1_l1_l_b_id(){return roads_[0].get_lanes()[0].get_left_boundary_id();}
```
please remove this

hamzaabbasi16 replied:
Removed!

ahmed-nad reviewed (commented):
1) Remove setters
2) Correct indentation
3) Use constructor initialization

muhammadhasnain12 committed:
extra code has removed

hamzaabbasi16 committed:
Update main.cpp

hamzaabbasi16 committed:
All of the required changes has been fixed!

ahmed-nad reviewed (commented):
Divide the classes into separate .h, .cpp 

Default constructor is available

hamzaabbasi16 committed:
Code splitted: in .h files

ahmed-nad commented on main.cpp:
```diff
@@ -7,126 +7,14 @@
-	}
 
-	// ********** Loading the lanes to the road structure ***********
+#include "Lane.h"
+#include "Road.h"
+#include "Map.h"
```
you only need to include map here

ahmed-nad reviewed (commented):
Divide the classes into .h and .cpp files separately please

ahmed-nad reviewed (commented):
Also please upload the cmake file as well

ahmed-nad commented on Lane.h:
```diff
@@ -0,0 +1,20 @@
+	float get_right_boundary_id(){return right_boundary_id;}
+	std::vector<std::vector<double>> get_centre_point(){return centre_point;}
+};
+Lane :: Lane(float a, float b, float c, std::vector<std::vector<double>> d)
+{
+	id = a;
+	left_boundary_id = b;
+	right_boundary_id = c;
+	centre_point = d;
```
you can remove these default constructors as they are not necessary

hamzaabbasi16 replied:
then how'd we initialize the data while instantiating the object of Lane and Road?

ahmed-nad reviewed (commented):
remove default constructors as well they are not necessary

hamzaabbasi16 committed:
Changes made: classes splitted into .h, .cpp files

- Header guards has been added
- Include files for Map.cpp has been reduced to only required libraries

hamzaabbasi16 committed:
Update README.md

hamzaabbasi16 committed:
Merge branch 'master' into branch1
